### PR TITLE
Update openzeppelin to v5.0.0

### PR DESCRIPTION
**Pull request by andreivladbrg**
The reason why I've added the the erc20 mock here to is that the package does not include the mocks `dir` from `test` in core.

### CODE REVIEW

**Review by smol-ninja — APPROVED** [review]
See comment below.

File: test/Base.t.sol [hunk 1]
```diff
@@ -20,6 +19,7 @@ import { SablierV2Batch } from "src/SablierV2Batch.sol";
 import { SablierV2MerkleLockupFactory } from "src/SablierV2MerkleLockupFactory.sol";
 import { SablierV2MerkleLockupLL } from "src/SablierV2MerkleLockupLL.sol";
 
+import { ERC20Mock } from "./mocks/erc20/ERC20Mock.sol";
```
Comment (smol-ninja) [inline test/Base.t.sol:22]: > The reason why I've added the the erc20 mock here to is that the package does not include the mocks dir from test in core.

I just checked and `ERC20` mock is included in v2-core package. You can import it using 

```solidity
import { ERC20Mock } from "@sablier/v2-core/test/mocks/erc20/ERC20Mock.sol";
```

**Review by smol-ninja — CHANGES_REQUESTED** [review]

File: test/Base.t.sol (hunk 1 above)
Comment (andreivladbrg) [reply test/Base.t.sol:22]: The repo yes

Not the package https://www.npmjs.com/package/@sablier/v2-core?activeTab=code

We are currently installing it as a git repo but I think we are going to change this, once everything is released on core

Comment (smol-ninja) [reply test/Base.t.sol:22]: You are right. Out of curiosity, why did we decide to include only `utils` in the package? Is there a Github discussion that I can read to understand the rationale behind it?

Comment (andreivladbrg) [reply test/Base.t.sol:22]: for precompiles: https://github.com/sablier-labs/v2-core/blob/main/test/utils/Precompiles.sol

https://github.com/sablier-labs/v2-periphery/blob/8f4c3de99b56af81ef467b03d296c441f60f14f2/test/integration/Integration.t.sol#L4

Comment (smol-ninja) [reply test/Base.t.sol:22]: Got it. Thanks for explaining.

**Review by smol-ninja — APPROVED** [review]

**Comment by smol-ninja** [comment]
Should we merge this now @andreivladbrg?

**Comment by andreivladbrg** [comment]
yeah

**Comment by PaulRBerg** [comment]
I have the same question as [here](https://github.com/sablier-labs/v2-core/pull/806#issuecomment-2037928508).

**Pull request merged**